services/predictors/vila/server/api_model.py

```diff
@@ -0,0 +1,42 @@
+"""
+Pydantic equivalents of classes from mmda.types
+"""
+
+from typing import List, Optional
+from pydantic import BaseModel, Field, validator
+
+from mmda.types.document import Document as MmdaDoc, SpanGroup as MmdaSpanGroup
+
+
+class Span(BaseModel):
+    start: int
+    end: int
+    box: Optional[List[float]] = Field(max_items=5, min_items=5)
+
+class BoxGroup(BaseModel):
+    boxes: List[List[float]]
+    type: Optional[str]
+
+    @validator("boxes", each_item=True)
+    def check_box(cls, v):
+        assert len(v) == 5, "Box must have format [x, y, w, h, page]"
+        assert int(v[4]) == v[4], "Page must be an integer value"
+
+
+class SpanGroup(BaseModel):
+    spans: List[Span]
+    type: Optional[str]
+
+    @classmethod
+    def from_mmda(cls, grp: MmdaSpanGroup):
```

Comment:
Convert back and forth between `mmda` types.

Reply:
hmm, a bit unfortunate that this is necessary but I guess fine for now. open to ideas about how to consolidate.